### PR TITLE
SD Elements: Docker Non Root User

### PR DESCRIPTION
**Pull request by anthonyvelazquez**
# SD Elements: Docker Non Root User 

### What was the problem?

`root` is used by default for all the Dockerfile commands

### How does this fix it?

Create a new non root user to comply with SD Elements request

### CODE REVIEW

**Review by au2 — CHANGES_REQUESTED** [review]
1) You need to fix all the Dockerfiles (including the java ones). Should not be much different. If you want to do those in another pr that is fine too.
2) Please fix the checks. build-docker-images is failing. Something to do with python version.

**Review by jbregstone — APPROVED** [review]
I will approve but I did not try running as a rootless user

File: service-data-access/src/docker/Dockerfile [hunk 1]
```diff
@@ -6,4 +6,5 @@ WORKDIR /project
 COPY ${ENTRYPOINT_FILE} entrypoint.sh
 RUN chmod +x entrypoint.sh
 COPY ${JAR_FILE} service-data-access.jar
+RUN adduser -m docker
 CMD ["sh", "entrypoint.sh"]
```
Comment (au2) [inline service-data-access/src/docker/Dockerfile:10]: We are only adding the user? I thought we needed to switch to the user as well?

File: app/src/docker/Dockerfile [hunk 2]
```diff
@@ -8,4 +8,5 @@ COPY ${JAR_FILE} starter-java.jar
 EXPOSE 8081
 HEALTHCHECK CMD curl --fail http://localhost:8081/health || exit 1
 LABEL org.opencontainers.image.source=https://github.com/department-of-veterans-affairs/abd-vro
+RUN adduser -m docker
```
Comment (au2) [inline app/src/docker/Dockerfile:11]: Only adding user? Shouldn't we switch as well?

**Review by au2 — CHANGES_REQUESTED** [review]

**Review by au2 — APPROVED** [review]
@anthonyvelazquez  I think we should return an error when there is no content but I do not think it will ever go to that line so will let it go. For future I would identify if that line is ever reached always return error when there is no content.

**Review by au2 — CHANGES_REQUESTED** [review]
I meant to approve the other pr which I did. This still needs the inline comments addressed.

**Review by au2 — APPROVED** [review]
End points are working. Moving this.

**Pull request merged**